engine/flink/api/src/main/scala/pl/touk/nussknacker/engine/flink/api/process/FlinkSource.scala

```diff
@@ -17,24 +17,34 @@ import scala.reflect._
   * Source with methods specific for Flink
   * @tparam T - type of event that is generated by this source. This is needed to handle e.g. syntax suggestions in UI
   */
+//TODO: remove type T parameter
 trait FlinkSource[T] extends Source[T] {
 
   def sourceStream(env: StreamExecutionEnvironment,
                    flinkNodeContext: FlinkCustomNodeContext): DataStream[Context]
 
 }
 
-trait SourceTestSupport[T] { self: Source[T] =>
+
+trait FlinkIntermediateRawSource[T] {
```

Comment:
Can you write some notice about purpose of this class? I know that it will be hard, but IMO we should try. Name is as good as it can be, but it is quite hard to explain purpose without comment in this case.

Reply:
I wrote some comment. Please check if it is sufficient.